src/Scrolled Window/main.vala

```diff
@@ -0,0 +1,109 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+private bool auto_scrolling = false;
+
+public void main () {
+    var scrolled_window = (Gtk.ScrolledWindow) workbench.builder.get_object ("scrolled_window");
+    var container = (Gtk.Box) workbench.builder.get_object ("container");
+    var toggle_orientation = (Gtk.ToggleButton) workbench.builder.get_object ("toggle_orientation");
+    var button_start = (Gtk.Button) workbench.builder.get_object ("button_start");
+    var button_end = (Gtk.Button) workbench.builder.get_object ("button_end");
+
+    button_start.sensitive = false;
+
+    HashTable<Gtk.Orientation, Gtk.Scrollbar> scrollbars = new HashTable<Gtk.Orientation, Gtk.Scrollbar> (direct_hash, direct_equal);
+    scrollbars[Gtk.Orientation.HORIZONTAL] = (Gtk.Scrollbar) scrolled_window.get_hscrollbar ();
+    scrollbars[Gtk.Orientation.VERTICAL] = (Gtk.Scrollbar) scrolled_window.get_vscrollbar ();
+
+    toggle_orientation.toggled.connect (() => {
+        if (toggle_orientation.active) {
+            container.orientation = Gtk.Orientation.HORIZONTAL;
+        } else {
+            container.orientation = Gtk.Orientation.VERTICAL;
+        }
+    });
+
+    int num_items = 20;
+    for (int i = 0; i < num_items; i++) {
+        populate_container (container, @"Item $(i + 1)");
+    }
+
+    scrollbars.foreach ((orientation, scrollbar) => {
```

Comment:
Please iterate over the hash table using its list of keys rather than the foreach method, like this:

```vala
foreach (Gtk.Orientation orientation in scrollbars.get_keys ()) {
    Gtk.Scrollbar scrollbar = scrollbars[orientation];
// ...
}
```